### DISCUSSION TITLE
React 19 does not contain `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED`

### DISCUSSION THREAD

**Discussion by simPod** [tailwindlabs/headlessui#3149] [Ideas]
https://github.com/tailwindlabs/headlessui/blame/afc9cb648b1f1925694510073501098266737078/packages/%40headlessui-react/src/utils/stable-collection.tsx#L66

We should consider migrating to `__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE`

https://github.com/facebook/react/pull/28789

**Comment by kms0219kms** [comment]
Later issue but have been solved from #3167
